# QS-ENN — Cross-Dataset Experimental Results

Final aggregate results for the ten medical datasets evaluated with the fold-isolated QS-ENN experimental protocol.

## Table 1. Medical datasets after leakage control and fold-isolated preprocessing

In [ ]:
import pandas as pd
table1 = pd.read_csv("../results/TABLE1.csv")
display(table1)


Dataset,n,Class distribution,IR,Var. (%)
Heart Disease,920,509 : 411,1.24,64.1 ± 0.3
Typhoid,1064,599 : 465,1.29,74.5 ± 0.3
Liver Disease,583,416 : 167,2.49,92.7 ± 0.3
PIMA Indian Diabetes,768,500 : 268,1.87,83.5 ± 0.7
Frankfurt Hospital Diabetes,2000,1316 : 684,1.92,83.9 ± 0.2
Breast Cancer,569,357 : 212,1.68,87.9 ± 0.3
Obesity (4 classes),108,29 : 12 : 20 : 47,3.92,100.0 ± 0.0
Lung Cancer,309,270 : 39,6.92,62.1 ± 0.5
HCV (Hepatitis C),615,540 : 75,7.20,90.6 ± 0.5
Cervical Cancer,858,803 : 55,14.60,78.9 ± 0.6


## Table 2. QS-ENN performance across ten medical datasets

For binary datasets, Recall, Precision, and F1-score refer to the minority class. For Obesity, macro-averaged values are reported.

In [ ]:
table2 = pd.read_csv("../results/TABLE2.csv")
display(table2.round(3))


Dataset,Recall,Precision,F1-score,G-mean,AUC,Accuracy
Heart Disease,0.813,0.708,0.754,0.764,0.806,0.763
Typhoid,0.586,0.425,0.492,0.473,0.495,0.474
Liver Disease,0.618,0.384,0.472,0.606,0.646,0.607
PIMA Indian Diabetes,0.747,0.539,0.624,0.696,0.738,0.685
Frankfurt Hospital Diabetes,0.865,0.645,0.738,0.805,0.859,0.790
Breast Cancer,0.954,0.922,0.937,0.952,0.972,0.952
Obesity (4 classes),0.664,0.631,0.622,0.386,0.870,0.685
Lung Cancer,0.758,0.477,0.572,0.800,0.845,0.856
HCV (Hepatitis C),0.504,0.645,0.554,0.675,0.765,0.907
Cervical Cancer,0.130,0.180,0.140,0.253,0.580,0.900


## Table 3. F1-score comparison of QKNN resampling methods across ten medical datasets

† Typhoid is reported for documentation only and is excluded from comparative and ablation claims because AUC=0.495.

In [ ]:
table3 = pd.read_csv("../results/TABLE3.csv")
display(table3.round(3))


Dataset,No res.,SMOTE,B-SMOTE,SMOTE-ENN,SMOTE-Tomek,Q-SMOTE,Q-SMOTEV2,QS-ENN
Heart Disease,0.752,0.753,0.751,0.763,0.754,0.747,0.751,0.754
Typhoid †,0.394,0.423,0.430,0.477,0.423,0.400,0.404,0.492
Liver Disease,0.373,0.467,0.453,0.494,0.473,0.486,0.480,0.472
PIMA Indian Diabetes,0.541,0.578,0.569,0.636,0.587,0.584,0.581,0.624
Frankfurt Hospital Diabetes,0.798,0.851,0.851,0.727,0.851,0.872,0.878,0.738
Breast Cancer,0.936,0.934,0.921,0.944,0.936,0.919,0.920,0.937
Obesity (4 classes),0.660,0.686,0.705,0.605,0.663,0.664,0.695,0.622
Lung Cancer,0.544,0.602,0.588,0.580,0.605,0.534,0.601,0.572
HCV (Hepatitis C),0.545,0.482,0.506,0.470,0.481,0.437,0.494,0.554
Cervical Cancer,0.044,0.146,0.151,0.156,0.147,0.142,0.165,0.140


## Table 4. Cross-dataset F1-score ablation analysis of QS-ENN

Sign convention: ΔF1 = F1(ablated variant) − F1(complete QS-ENN). A positive value means removing or replacing the component increases F1.

In [ ]:
table4 = pd.read_csv("../results/TABLE4.csv")
display(table4)


Dataset,Without Pillar 1,Pillar 2 → neighbor,Without Pillar 3
Heart Disease,+0.0029,-0.0031,-0.0029
Typhoid †,—,—,—
Liver Disease,-0.0002,+0.0122,-0.0739
PIMA Indian Diabetes,-0.0046,+0.0019,-0.0491
Frankfurt Hospital Diabetes,+0.0064,+0.0213,+0.0815
Breast Cancer,-0.0006,-0.0154,+0.0004
Obesity (4 classes),+0.0199,+0.0369,+0.0073
Lung Cancer,-0.0081,+0.0071,+0.0194
HCV (Hepatitis C),-0.0434,-0.0726,+0.0046
Cervical Cancer,+0.0040,+0.0020,-0.0333


## Table 5. Paired F1-score comparison of QS-ENN with three principal resampling baselines

* Holm-adjusted p<0.05; Δ values are based on unrounded paired fold-level results. † Typhoid is excluded from comparative and ablation claims (AUC=0.495).

In [ ]:
table5 = pd.read_csv("../results/TABLE5.csv")
display(table5)


Dataset,QS-ENN F1,Δ vs SMOTE-ENN (p),Δ vs Q-SMOTE (p),Δ vs Q-SMOTEV2 (p),Interpretation
Heart Disease,0.754,-0.009 (0.639),+0.007 (0.639),+0.004 (0.655),No significant difference
Typhoid †,0.492,—,—,—,Excluded from comparative claims (AUC 0.495)
Liver Disease,0.472,-0.022 (0.281),-0.014 (0.742),-0.009 (0.742),No significant difference
PIMA Indian Diabetes,0.624,-0.012 (0.271),+0.040 (<0.001)*,+0.044 (<0.001)*,QS-ENN > both quantum oversamplers
Frankfurt Hospital Diabetes,0.738,+0.011 (0.080),-0.134 (<0.001)*,-0.139 (<0.001)*,Q-SMOTE and Q-SMOTEV2 superior
Breast Cancer,0.937,-0.007 (0.277),+0.018 (0.005)*,+0.017 (0.003)*,QS-ENN > both quantum oversamplers
Obesity (4 classes),0.622,+0.017 (0.673),-0.042 (0.126),-0.073 (0.033)*,Q-SMOTEV2 superior
Lung Cancer,0.572,-0.008 (0.761),+0.038 (0.377),-0.029 (0.411),No significant difference
HCV (Hepatitis C),0.554,+0.085 (0.012)*,+0.117 (<0.001)*,+0.060 (0.012)*,QS-ENN superior to all three
Cervical Cancer,0.140,-0.015 (1.000),-0.002 (1.000),-0.025 (0.458),No significant difference


## Applicability diagnostics

The strongest tested leave-one-dataset-out sign predictor achieved 6/8 correct decisions; the majority-sign reference achieved 5/8.

In [ ]:
diagnostics = pd.read_csv("../results/applicability_diagnostics.csv")
display(diagnostics.round(3))


Descriptor,Spearman ρ,p-value
Fraction of synthetic samples moving toward the majority centroid,0.214,0.610
Mean centroid gain,0.095,0.823


## Table 6. Equal-budget nested-CV fairness analysis across eight binary datasets

In [ ]:
table6 = pd.read_csv("../results/TABLE6.csv")
display(table6.round(3))


Dataset,Fixed F1,Nested F1,Fixed rank,Nested rank
Breast Cancer,0.937,0.934,2,3
Cervical Cancer,0.140,0.180,7,1
Frankfurt Hospital Diabetes,0.738,0.742,7,7
HCV (Hepatitis C),0.554,0.616,1,1
Heart Disease,0.754,0.764,2,5
Liver Disease,0.472,0.469,5,6
Lung Cancer,0.572,0.592,6,2
PIMA Indian Diabetes,0.624,0.626,2,2


## Cross-dataset summary

The results are dataset-dependent. HCV provides the clearest broad QS-ENN advantage. PIMA and Breast Cancer outperform both quantum oversamplers, whereas Frankfurt Hospital Diabetes and Obesity favor Quantum-SMOTEV2. The ablation results show component-level sign reversals across datasets.